### PR TITLE
Fix omni directional cylindrical test failing

### PR DESCRIPTION
**Pull request by TheMangalex**
**Description**
Fixes a failing test due to #6396

I changed the mIsOmniDirectional variable back to being horizontalFOV > M_PI and instead reprojected all back points to fix the issue from #6395

I additionally added another object to the recognition test which is perpendicular to the sensor plane to ensure that these objects are also correctly recognized. 

I want to add another test which checks that the recognized position is correct as this was a problem in #6395

**Related Issues**
This pull-request addresses issue #6395

**Screenshots**
Newly added object for test
image

Fixed detection
image

### CODE REVIEW

File: tests/api/controllers/camera_recognition/camera_recognition.c [hunk 1]
```diff
@@ -9,15 +9,15 @@
...
+#define VISIBLE_SOLID_NUMBER 7
 
 // This test is mainly testing the functionalities for a planar camera.
 // Some basic tests for spherical and cylindrical cameras are also performed checking mainly
 // the number of visible objects and the position on camera.
 
 // objects visible in planar camera
 static const char *visible_solid_models[VISIBLE_SOLID_NUMBER] = {
-  "visible sphere", "visible box", "sub solid", "visible capsule", "composed solid", "visible sphere without BO"};
+  "visible sphere", "visible box", "sub solid", "visible capsule", "composed solid", "visible sphere without BO", "perpendicular box"};
```
Comment (omichel) [inline tests/api/controllers/camera_recognition/camera_recognition.c:20]: clang format fix:
```suggestion
  "visible sphere", "visible box", "sub solid", "visible capsule", "composed solid", "visible sphere without BO",
  "perpendicular box"};
```

File: tests/api/controllers/camera_recognition/camera_recognition.c [hunk 2]
```diff
@@ -45,12 +45,14 @@ int main(int argc, char **argv) {
                       VISIBLE_SOLID_NUMBER + 1, object_number);
 
   object_number = wb_camera_recognition_get_number_of_objects(camera_spherical);
-  ts_assert_int_equal(object_number, 10, "The spherical camera should initially see %d objects and not %d (with occlusion).", 9,
-                      object_number);
+  ts_assert_int_equal(object_number, VISIBLE_SOLID_NUMBER + 4,
+    "The spherical camera should initially see %d objects and not %d (with occlusion).", VISIBLE_SOLID_NUMBER + 4,
+    object_number);
```
Comment (omichel) [inline tests/api/controllers/camera_recognition/camera_recognition.c:50]: clang format:
```suggestion
  ts_assert_int_equal(object_number, VISIBLE_SOLID_NUMBER + 4,
                      "The spherical camera should initially see %d objects and not %d (with occlusion).",
                      VISIBLE_SOLID_NUMBER + 4, object_number);
```

File: tests/api/controllers/camera_recognition/camera_recognition.c [hunk 3]
```diff
@@ -45,12 +45,14 @@ int main(int argc, char **argv) {
...
 
   object_number = wb_camera_recognition_get_number_of_objects(camera_cylindrical);
-  ts_assert_int_equal(object_number, 8, "The cylindrical camera should initially see %d objects and not %d (with occlusion).",
-                      7, object_number);
+  ts_assert_int_equal(object_number, VISIBLE_SOLID_NUMBER + 2,
+    "The cylindrical camera should initially see %d objects and not %d (with occlusion).",
+    VISIBLE_SOLID_NUMBER + 2, object_number);
```
Comment (omichel) [inline tests/api/controllers/camera_recognition/camera_recognition.c:55]: clang format fix:
```suggestion
  ts_assert_int_equal(object_number, VISIBLE_SOLID_NUMBER + 2,
                      "The cylindrical camera should initially see %d objects and not %d (with occlusion).",
                      VISIBLE_SOLID_NUMBER + 2, object_number);
```

**Review by omichel — CHANGES_REQUESTED** [review]
We should fix clang format errors.

**Comment by omichel** [comment]
It seems that there are more clang format issues.

**Comment by TheMangalex** [comment]
I did another update to the clang formatting. This time it seems all test were running successful.

**Comment by omichel** [comment]
Thank you. Is the PR ready for review? (It is still marked as Draft.)

**Comment by TheMangalex** [comment]
Yes, I updated it now.

**Review by omichel — APPROVED** [review]
That looks great to me.
Thank you.

**Pull request merged**